compiler/src/iree/compiler/Codegen/Common/FastMathPatterns.cpp

```diff
@@ -19,109 +21,114 @@ namespace {
 struct FastErfPattern : public OpRewritePattern<math::ErfOp> {
   using OpRewritePattern::OpRewritePattern;
 
+  // Helper to evaluate a polynomial using FMA chains for both scalar and vector
+  // types.
+  static Value evalErfPolynomial(Value x, Value t, ArrayRef<Value> coeffs,
+                                 RewriterBase &rewriter, Location loc) {
+    Value acc = coeffs[0];
+    for (size_t i = 1; i < coeffs.size(); ++i) {
+      acc = rewriter.create<math::FmaOp>(loc, t, acc, coeffs[i]);
+    }
+    return rewriter.create<math::FmaOp>(loc, x, acc, x);
+  }
+
   LogicalResult matchAndRewrite(math::ErfOp op,
                                 PatternRewriter &rewriter) const override {
     Location loc = op.getLoc();
     Value input = op.getOperand();
     Type resultType = op.getType();
 
-    // Erf only supports f32.
-    if (!resultType.isF32()) {
-      return rewriter.notifyMatchFailure(op, "Result only supports f32");
+    // Only support f32 and vector types with f32 element type.
+    VectorType resVecType = llvm::dyn_cast<VectorType>(resultType);
+    if (!(resultType.isF32() ||
+          (resVecType && resVecType.getElementType().isF32()))) {
+      return rewriter.notifyMatchFailure(
+          op, "Result only supports f32 or vector types with f32 element type");
```

Comment:
> Also, to match error message styles commonly produced by other tools, start the first sentence with a lowercase letter, and finish the last sentence without a period, if it would end in one otherwise.

Given you're touching the code, can you help fix the style? 

```suggestion
          op, "result only supports f32 or vector types with f32 element type");
```

https://llvm.org/docs/CodingStandards.html#error-and-warning-messages

[optional] If it were me, I'd delete the above comment because we already document it in the error message.